### PR TITLE
dd: fix output issues

### PR DESCRIPTION
**Pull request by patricksjackson**
This change fixes multiple issues with the output of dd, that were easiest to complete in a single PR.

1. Correctly respect SIGUSR1 signals, by marking the setting as 'seen' once we have responded to it.
2. Fix a race condition in the output by moving all output into a single thread.
3. Fixes the bad combination of an existing 'progress' line and final output which can result in printing both on the same line.
4. Moves the check for whether output is needed (only every 1 second) into the main thread to reduce thread-to-thread communication.

The performance uplift of fixing the overzealous thread communication is significant for small blocksizes, and stacks with the buffer changes in #3600.

  | bs=4k count=1000000 | bs=1M count=20000 | bs=1G count=10
-- | -- | -- | --
Coreutils dd | 2.14s | 1.98s | 5.98s
Coreutils dd (Reused buffer #3600) | 1.99s | 1.60s | 1.95s
Coreutils dd (this change, output updates) | 1.29s | 1.79s | 5.96s
Coreutils dd (buffer+output) | 1.15s | 1.42s | 1.90s
GNU dd | 1.15s | 1.07s | 1.13s

### CODE REVIEW

File: src/uu/dd/src/progress.rs [hunk 1]
```diff
@@ -183,7 +194,10 @@ impl ProgUpdate {
     /// assert_eq!(iter.next().unwrap(), b"");
     /// assert!(iter.next().is_none());
     /// ```
-    fn write_transfer_stats(&self, w: &mut impl Write) -> std::io::Result<()> {
+    fn write_transfer_stats(&self, w: &mut impl Write, new_line: bool) -> std::io::Result<()> {
```
Comment (jfinkels) [inline src/uu/dd/src/progress.rs:197]: The documentation and the example code should be updated for this new parameter.

**Review by jfinkels — APPROVED** [review]
Cool

File: src/uu/dd/src/progress.rs (hunk 1 above)
Comment (patricksjackson) [reply src/uu/dd/src/progress.rs:198]: Thanks, done.

**Comment by sylvestre** [comment]
Looks like my manual merge didn't work, could you please fix it? sorry :(

**Comment by patricksjackson** [comment]
No problem @sylvestre, I've merged the current main branch into this one.

**Comment by sylvestre** [comment]
thanks!

**Comment by sylvestre** [comment]
@patricksjackson did you push it?

**Comment by patricksjackson** [comment]
I misunderstood, and I hadn't realized that the merge broke the tests.

I've fixed the tests that were broken due to the merge and have pushed the new changes.

**Pull request merged**